gunrock/app/rw/rw_enactor.cuh

```diff
@@ -0,0 +1,385 @@
+            return true;
+        };
+
+        cudaError_t retval = BaseIterationLoop:: template ExpandIncomingBase
+            <NUM_VERTEX_ASSOCIATES, NUM_VALUE__ASSOCIATES>
+            (received_length, peer_, expand_op);
+        return retval;
+    }
+}; // end of RWIteration
+
+/**
+ * @brief Template enactor class.
+ * @tparam _Problem Problem type we process on
+ * @tparam ARRAY_FLAG Flags for util::Array1D used in the enactor
+ * @tparam cudaHostRegisterFlag Flags for util::Array1D used in the enactor
+ */
+template <
+    typename _Problem,
+    util::ArrayFlag ARRAY_FLAG = util::ARRAY_NONE,
+    unsigned int cudaHostRegisterFlag = cudaHostRegisterDefault>
+class Enactor :
+    public EnactorBase<
+        typename _Problem::GraphT,
+        typename _Problem::GraphT::VertexT,
+        typename _Problem::GraphT::ValueT,
+        ARRAY_FLAG, cudaHostRegisterFlag>
+{
+public:
+    typedef _Problem                   Problem ;
+    typedef typename Problem::SizeT    SizeT   ;
+    typedef typename Problem::VertexT  VertexT ;
+    typedef typename Problem::GraphT   GraphT  ;
+    typedef typename GraphT::VertexT   LabelT  ;
+    typedef typename GraphT::ValueT    ValueT  ;
+    typedef EnactorBase<GraphT, LabelT, ValueT, ARRAY_FLAG, cudaHostRegisterFlag>
+        BaseEnactor;
+    typedef Enactor<Problem, ARRAY_FLAG, cudaHostRegisterFlag> 
+        EnactorT;
+    typedef RWIterationLoop<EnactorT> 
+        IterationT;
+
+    Problem *problem;
+    IterationT *iterations;
+
+    /**
+     * @brief hello constructor
+     */
+    Enactor() :
+        BaseEnactor("RW"),
+        problem    (NULL  )
+    {
+        // <OPEN> change according to algorithmic needs
+        this -> max_num_vertex_associates = 0;
+        this -> max_num_value__associates = 1;
+        // </OPEN>
+    }
+
+    /**
+     * @brief hello destructor
+     */
+    virtual ~Enactor() { /*Release();*/ }
+
+    /*
+     * @brief Releasing allocated memory space
+     * @param target The location to release memory from
+     * \return cudaError_t error message(s), if any
+     */
+    cudaError_t Release(util::Location target = util::LOCATION_ALL)
+    {
+        cudaError_t retval = cudaSuccess;
+        GUARD_CU(BaseEnactor::Release(target));
+        delete []iterations; iterations = NULL;
+        problem = NULL;
+        return retval;
+    }
+
+    /**
+     * @brief Initialize the problem.
+     * @param[in] problem The problem object.
+     * @param[in] target Target location of data
+     * \return cudaError_t error message(s), if any
+     */
+    cudaError_t Init(
+        Problem          &problem,
+        util::Location    target = util::DEVICE)
+    {
+        cudaError_t retval = cudaSuccess;
+        this->problem = &problem;
+
+        // Lazy initialization
+        GUARD_CU(BaseEnactor::Init(
+            problem, Enactor_None,
+            // <OPEN> change to how many frontier queues, and their types
+            2, NULL,
```

Comment:
RW only uses a single vertex frontier, so should be `1, NULL,` here

Reply:
resolved